lint/src/main/java/com/firebaseui/lint/NonGlobalIdDetector.kt

```diff
@@ -0,0 +1,52 @@
+package com.firebaseui.lint
+
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.LayoutDetector
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.XmlContext
+import org.w3c.dom.Element
+
+/**
+ * Lint detector to find layout attributes that reference "@id" where they should reference
+ * "@+id" instead.
+ */
+class NonGlobalIdDetector : LayoutDetector() {
+    override fun getApplicableElements() = Detector.XmlScanner.ALL
+
+    override fun visitElement(context: XmlContext, element: Element) {
+        (0 until element.attributes.length)
+                .map { element.attributes.item(it) }
+                .filter { it.nodeValue.contains(NON_GLOBAL_ID_PREFIX) }
+                .forEach {
+                    val quickFix = fix()
+                            .replace()
+                            .name("Fix id")
+                            .text(it.nodeValue)
+                            .with(it.nodeValue.replace(NON_GLOBAL_ID_PREFIX, GLOBAL_ID_PREFIX))
+                            .build()
+
+                    context.report(NON_GLOBAL_ID, it, context.getLocation(it), REPORT_MESSAGE, quickFix)
```

Comment:
I would prefer this like so, but it's up to you:
```kotlin
context.report(NON_GLOBAL_ID, it, context.getLocation(it), REPORT_MESSAGE, fix()
        .replace()
        .name("Fix id")
        .text(it.nodeValue)
        .with(it.nodeValue.replace(NON_GLOBAL_ID_PREFIX, GLOBAL_ID_PREFIX))
        .build()
)
```

Reply:
Not a strong opinion either

Reply:
SGTM.